feat(chip): align atom with design system

andresito87 opened the pull request:
## Summary

- Completa el componente `Chip` y lo alinea con los patrones actuales del Design System.
- Corrige accesibilidad, contraste light/dark, interacción clickable/closable y publicación del componente en la API pública.
- Añade stories y tests para cubrir variantes, estados y regresiones del componente.

Closes #19

## Type of change

- [x] 🧩 New component
- [ ] 🐛 Bug fix
- [ ] 🎨 Design tokens
- [x] ♿ Accessibility
- [ ] 🏗️ Infrastructure
- [x] 📚 Documentation

## Component checklist (skip if not applicable)

- [x] Follows the 5-file structure (`types.ts`, `use*.ts`, `Component.tsx`, `index.ts`, `*.stories.tsx`)
- [x] CVA variants defined in `types.ts`, not inline
- [x] No hardcoded colors — uses tokens from `theme.css`
- [x] No `any` types — TypeScript strict
- [x] ARIA attributes present and correct
- [x] Keyboard navigable (Tab, Enter, Escape where applicable)
- [x] Dark mode works correctly
- [x] Storybook stories cover: default, variants, states (hover, focus, disabled)
- [x] Tests added or updated

## How to test

1. `pnpm run storybook`
2. Navegar a `Atoms/Chip`
3. Validar variantes, tamaños, estados interactivos y dark mode:
   - `Size`
   - `Color`
   - `Variant`
   - `Clickable`
   - `Closable And Clickable`
   - `Disabled And Closable`
   - `Dot With Text`
4. Confirmar que el addon de accessibility no reporte problemas reales del componente.
5. Ejecutar `pnpm test -- src/components/atoms/chip/Chip.test.tsx`

## Screenshots / recordings (if applicable)

- Se validó visualmente durante la corrección iterativa en Storybook para light y dark mode.

## Notes for reviewer

- Cuando `closable` se combina con interacción (`onClick`/`selectable`), el componente usa un patrón de split-actions para evitar controles interactivos anidados.
- Se añadió export público en `src/index.ts`.
- El issue #19 actualmente no muestra label de aprobación; si el workflow exige aprobación explícita del issue, puede bloquear checks hasta que se añada.

andresito87 committed:
[ADD] Chip component

egdev6 commented:
@andresito87 varias cosas:

VISUALIZACIÓN

1- La variante shadow me queda rara con esa sombra en light mode
<img width="1056" height="172" alt="SCR-20250825-kltl" src="https://github.com/user-attachments/assets/af7bc6c1-d98a-4609-92d9-d7e0b98ac6a4" />

2-Asegúrate que los elementos cojan los colores correctos en light mode
<img width="227" height="89" alt="SCR-20250825-kmhh" src="https://github.com/user-attachments/assets/1e2ee2f3-a9d3-495b-8dd8-a6cddee873d5" />

3-Me queda raro que el icono de close sea tan pequeño (sin hover queda como muy separado) quizás sobreescribir los estilos para que no añada ese padding y el círculo en el hover)
<img width="437" height="81" alt="SCR-20250825-knca" src="https://github.com/user-attachments/assets/962f4093-a979-4464-b87f-7f22aabdb0b5" />


4-El list lo eliminaría para hacerlo molécula en el futuro (aunque es un buen aporte), ya que lo veo más propio de una combinación de varias cosas (noi sólo el chip)
<img width="1058" height="299" alt="SCR-20250825-knvd" src="https://github.com/user-attachments/assets/c85bc003-1230-4cd7-ad25-a6f404867ebd" />


5-Le quitaría los efectos de hover a todas las variantes no interactuables con click ya que el componente en sí queda raro que tenga ese fecto hover cuando sólo es un componente visual e informativo.

Lo demás lo veo guay. Comentarios del código... en la lineas afectadas

andresito87 committed:
[FIX] Chip component Shadow Variant

andresito87 committed:
[FIX] Deleting Chip adding list component and giving more size to close icon in chip component

andresito87 committed:
[FIX] Chip Component -Deleting hover effect in every non-interactable elements

andresito87 committed:
[FIX] Chip component - fixing accessibility problems

andresito87 committed:
[FIX]: Chip variants styles

andresito87 committed:
fix(chip): align accessibility, variants, and stories

andresito87 committed:
fix(chip): address review feedback

andresito87 commented:
Applied a final pass to address the remaining review feedback and align the Chip atom with current project conventions.

### What was updated
- replaced the local `cn` helper in `useChip.ts` with the shared utility from `@/lib/utils`
- removed the extra `gap-0.5` override so spacing continues to come from the size variants
- replaced `ring-accent` usages with token-based Tailwind classes consistent with the rest of the component
- removed the misleading leftover comment in `types.ts`
- fixed the `Stress` story so `Icon` is no longer passed unsupported DOM/ARIA props directly
- updated the avatar scale utility to the canonical Tailwind v4 CSS variable syntax to avoid IDE warnings and keep the codebase aligned with modern Tailwind conventions
- removed the previous `ref as any` approach and tightened the root typing around the `button` / `div` render paths

### Validation
- `pnpm exec tsc --noEmit` ✅
- `pnpm test -- src/components/atoms/chip/Chip.test.tsx` ✅

### Notes
- public export remains correctly exposed through `src/index.ts` and the local chip barrel export
- split-actions behavior for `closable + interactive` was preserved so we still avoid nested interactive controls

This should leave the PR in a cleaner state both for review and for long-term maintainability.

egdev6 committed:
fix(chip): address final audit issues

egdev6 committed:
fix(a11y): resolve storybook audit failures
